ptf/saiacl.py

```diff
@@ -0,0 +1,5852 @@
+                      self.port4, self.port5, self.port6, self.port7,
+                      self.port8, self.port9, self.port10, self.port11,
+                      self.port12, self.port13, self.port14, self.port15,
+                      self.port16, self.port17, self.port18, self.port19,
+                      self.port20, self.port21, self.port22, self.port23,
+                      self.port24, self.port25, self.port26, self.port27,
+                      self.port28, self.port29, self.port30, self.port31]
+
+        for index, port in enumerate(ports_list):
+            # create ACL table group
+            acl_table_group_ingress = sai_thrift_create_acl_table_group(
+                self.client,
+                acl_stage=group_stage_ingress,
+                acl_bind_point_type_list=group_bind_point_type_list,
+                type=group_type)
+
+            print(index, port)
+            acl_ingress_groups_list.append(acl_table_group_ingress)
+            # bind this ACL group to ports object id
+            print("Bind ACL ingress group 0x % lx to port 0x % lx" % (
+                acl_ingress_groups_list[index], port))
+            sai_thrift_set_port_attribute(
+                self.client, port,
+                ingress_acl=acl_ingress_groups_list[index])
+
+        table_stage_ingress = SAI_ACL_STAGE_INGRESS
+        table_bind_point_list = [SAI_ACL_BIND_POINT_TYPE_PORT]
+        table_bind_point_type_list = sai_thrift_s32_list_t(
+            count=len(table_bind_point_list), int32list=table_bind_point_list)
+
+        packet_action = sai_thrift_acl_action_data_t(
+            parameter=sai_thrift_acl_action_parameter_t(
+                s32=SAI_PACKET_ACTION_DROP))
+
+        print("Create ACL field data")
+        src_ip_t_ipv4 = sai_thrift_acl_field_data_t(
+            data=sai_thrift_acl_field_data_data_t(ip4=ipv4_addr),
+            mask=sai_thrift_acl_field_data_mask_t(ip4=ip_mask))
+
+        # create ACL tables
+        print("Create ACL tables")
+        acl_ingress_ipv4_table_id2 = sai_thrift_create_acl_table(
+            self.client,
+            acl_stage=table_stage_ingress,
+            acl_bind_point_type_list=table_bind_point_type_list,
+            field_src_ip=True)
+
+        acl_ingress_mirror_table_id = sai_thrift_create_acl_table(
+            self.client,
+            acl_stage=table_stage_ingress,
+            acl_bind_point_type_list=table_bind_point_type_list,
+            field_src_ip=True,
+            field_dscp=True)
+
+        acl_ingress_ipv4_table_id1 = sai_thrift_create_acl_table(
+            self.client,
+            acl_stage=table_stage_ingress,
+            acl_bind_point_type_list=table_bind_point_type_list,
+            field_src_ip=True)
+
+        # create ACL ingress table group members
+        for acl_group in acl_ingress_groups_list:
+            # create ACL table group member 2 - v4 tables
+            acl_group_ingress_ip_member_id2 = \
```

Comment:
Our SDE cannot support 2 V4 permit/deny ACL tables, could you check if this works?

Reply:
SonicACLTest works, and in bf_switch I can see a few IPV4 ACL tables at the same time.

Reply:
Yeah, that is wrong. We need to modify the test.